### DISCUSSION TITLE
Video files : download or open ?

### DISCUSSION THREAD

**Discussion by rockrider69** [rejetto/hfs#401] [Q&A] [answered]
Hi , 

I have been using HFS V2 for some time now, and strictly to download my video files when I am on my way and watch later (not streaming).

Just starting to play with V3 now,  I see that if I select a video files I can either Download or Open.
Does Open means some kind of streaming is available now ?

At least based on 0.50-Alpha3 if I try Open and select VLC, I get some quick error from VLC (not easy to catch the message...) and it does not start to play.
Download is fully ok though ;)

More to understand what to expect !
Thanks in advance

**Comment by rejetto** [comment, marked as answer]
hey!
"open" is just leaving the browser decide what to do with the link, while "download" is just forcing the download.
The browser POSSIBLY opens the file, if it supports by file format.
So, at the moment, the "streaming" capabilities are limited to what the browser already has to offer.
Is not easy to go beyond that. I've already studied some possibilities, but that's not the focus of the project, and I still have to reach final version with all basic features.

To stream a video that's not supported by the browser means to convert it on-the-fly to a format that IS supported.
There was a very cool opportunity to convert it in the browser itself, but has been stopped because of security issues.
What's left is converting on the server, that is tricky and heavy on the CPU.

**Reply by rockrider69** [reply]
Thanks a lot for the explanations, it makes sense that the local browser plays a key role in here !

Just for info.
First I was using my android phone with the Samsung/Chrome based browser, but standard Chrome is the same result. It seems more VLC complaining...
From the PC (Edge browser), the Open action first download the file, then automatically starts my prefered video PotPlayer. No issue here, video starts as expected.
In both cases no real streaming I understand, but as you say that is a tricky one.

I am now going to migrate to HFS V3, for all the security improvements (https and more ....).

Many thanks for HFS, please keep up the great work with this !

**Comment by rejetto** [comment]
thanks!
anyway, you are not getting the video streamed inside the browser because the browser doesn't support that file format.
beware, the file format is not given just by the file extension. An mp4 can contain h264 and h265 video (codec). h264 is supported by chrome, h265 is not.
